### PR TITLE
Rest endpoint for Remote Server Logging

### PR DESCRIPTION
**Pull request by Thumimku**
## Purpose
Fixes https://github.com/wso2/product-is/issues/16588

# Prerequestie

- [x] Need to merge https://github.com/wso2/carbon-commons/pull/476 
- [x] Bump the version

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java [hunk 1]
```diff
@@ -97,4 +99,23 @@ public void setCorsManagementService(CORSManagementService corsManagementService
...
     }
+
+    /**
+     * Get RemoteLoggingConfigService osgi service.
+     *
+     * @return RemoteLoggingConfigService
+     */
+    public RemoteLoggingConfigService getRemoteLoggingConfigService() {
+
+        return ConfigsServiceHolder.getInstance().remoteLoggingConfigService;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java:110]: shouldn't this be `this.remoteLoggingConfigService` ?

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/Constants.java [hunk 2]
```diff
@@ -105,7 +108,29 @@ public enum ErrorMessage {
                                                       "Private Key JWT Validation configuration."),
         ERROR_JWT_AUTHENTICATOR_SERVICE_NOT_FOUND("60505",
                 "Private Key JWT Authenticator is not supported.",
-                "Private Key JWT Authenticator service is unavailable at the moment.");
+                "Private Key JWT Authenticator service is unavailable at the moment."),
+
+        /**
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/Constants.java:113]: this is a doc comment, which is not required here.

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/factory/RemoteLoggingOGSiServiceFactory.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/factory/RemoteLoggingOGSiServiceFactory.java:2]: License is not correct.

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java (hunk 1 above)
Comment (Thumimku) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java]: I think both are fine, but the suggestion is more appropriate.

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java [hunk 4]
```diff
@@ -97,4 +99,23 @@ public void setCorsManagementService(CORSManagementService corsManagementService
...
+        return this.remoteLoggingConfigService;
+    }
+    /**
+     * Set RemoteLoggingConfigService osgi service.
+     *
+     * @param remoteLoggingConfigService RemoteLoggingConfigService.
+     */
+    public void setRemoteLoggingConfigService(RemoteLoggingConfigService remoteLoggingConfigService) {
+
+        ConfigsServiceHolder.getInstance().remoteLoggingConfigService = remoteLoggingConfigService;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java:119]:
```suggestion
        this.remoteLoggingConfigService = remoteLoggingConfigService;
```

**Review by madurangasiriwardena — APPROVED** [review]

**Pull request merged**